pkg/cloudprovider/cloud.go

```diff
@@ -69,6 +69,11 @@ func GetLoadBalancerName(service *v1.Service) string {
 	//GCE requires that the name of a load balancer starts with a lower case letter.
```

Comment:
Humm...In fact I'm surprised that we have such function in cloudprovider, I thought loadbalancer name would be cloud provider specific --- as @jhorwit2 mentioned, different clouds have different requirements, if this is going to be an API, what would the API validation looks like...

The definition of loadbalancer name is also obscure to me --- usually a LB is constructed by multiple LB resources in cloud, are they all supposed to use the same name? Or as a prefix?

More questions, should this annotation/API (if we are adding it) be mutable after service creation? Wouldn't this increase the risk of orphaning  LB resources if they can now have arbitrary names (I suppose LB cleanup will continue depending on the caching layer before the finalizer solution is in-place)?

cc @nicksardo

Reply:
for loadbalancer name i'm looking to change the implementation to use namespace+service name+custom name to prevent name collisions and this annotation should not be mutable after service creation, otherwise i'm open to suggestion